src/map.h

```diff
@@ -120,6 +120,7 @@ struct visibility_variables {
     bool variables_set = false;
     bool u_sight_impaired = false;
     bool u_is_boomered = false;
+    bool clairvoyance_field = false;
```

Comment:
If you want the with-Magiclysm version of this code to run faster you could cache the `field_type_id` here in addition to the `bool`, then it won't need to be converted every time in the inner loop.

Reply:
Probably makes sense to conflate the `bool` and value into an `optional`, as suggested by @Aivean [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/43469#issuecomment-687661215).